andromedocs-theme/gatsby-node.js

```diff
@@ -0,0 +1,49 @@
+const path = require("path");
+const { createFilePath } = require("gatsby-source-filesystem");
+
+exports.onCreateNode = ({ node, actions, getNode }) => {
+  const { createNodeField } = actions;
+
+  if (node.internal.type === "Mdx") {
+    const value = createFilePath({ node, getNode });
+
+    createNodeField({
+      name: "slug",
+      node,
+      value: `${value}`,
+    });
```

Comment:
This way you can get the slug in frontmatter if exists
```suggestion
    const slug =
      node.frontmatter.slug ||
      createFilePath({ node, getNode }) // later we need to pass a basePath


    createNodeField({
      name: "slug",
      node,
      value: slug,
    });
```